### PR TITLE
Surface mesh approximation: Fix demo crash

### LINKED ISSUE #3617: VSA demo crash
In the demo:

1. Load any file
2. Generate seeds
3. Try to un-hide the polygon soup or the patch polygons
4. ???
5. <s>Profit!!!</s> Crash!!!

Likely because they haven't been generated yet, but then they shouldn't be visible.

### PR DESCRIPTION
**Pull request by aboudev**
## Fix demo crash #3617 

Generate visualize items after meshing.

## Release Management

* Affected package(s): Surface mesh approximation
* Issue(s) solved (if any): fix #3617

### CODE REVIEW

**Comment by sloriot** [comment]
I don't know if you already did it but It would be nice to hide the widget and only display it when the action is called from the menu. @maxGimeno can probably help if you don't know how to do it.

**Comment by aboudev** [comment]
I think we've done that in commit 0cbb170f6e4619bb432e38f4ee9da192de6660cf.

**Comment by sloriot** [comment]
I think `dock_widget->setVisible(false);` in the `init()` function must be added.

**Comment by maxGimeno** [comment]
@sloriot more precisely, in the init() function there is 
>   dock_widget->setVisible(true);

that should be replaced by 
>   dock_widget->setVisible(false);

**Comment by aboudev** [comment]
The visibility of the widget was set in commit 493b1b5ec1c80c3a8704e70e24458d94da6c91b5 and is still in the code.
https://github.com/CGAL/cgal/blob/cb402f1e91e685383d2d5e333e12def2b6d9b91b/Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/Surface_mesh_approximation_plugin.cpp#L81

**Comment by maxGimeno** [comment]
Yes, you are right, I was looking at the previously stated commit. It should be good with the current code.

File: Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/Surface_mesh_approximation_plugin.cpp [hunk 1]
```diff
@@ -404,19 +365,18 @@ void Polyhedron_demo_surface_mesh_approximation_plugin::on_buttonFit_clicked() {
...
-
   sm_item->color_vector() = approx.proxy_colors();
   sm_item->setItemIsMulticolor(true);
   sm_item->computeItemColorVectorAutomatically(false);
   sm_item->invalidateOpenGLBuffers();
   scene->itemChanged(scene->item_id(sm_item));
 
   if (ui_widget.checkAutomatic->isChecked()) {
+    if (!approx.visual_items().has_meshing_items) {
+      mi->information(QString("Please meshing before checking auto meshing."));
```
Comment (maxGimeno) [inline Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/Surface_mesh_approximation_plugin.cpp]: "Please mesh" if I'm not mistaken.

File: Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/Surface_mesh_approximation_plugin.cpp [hunk 2]
```diff
@@ -456,19 +416,18 @@ void Polyhedron_demo_surface_mesh_approximation_plugin::on_buttonAdd_clicked() {
...
```
Comment (maxGimeno) [inline Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/Surface_mesh_approximation_plugin.cpp]: Same here

File: Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/Surface_mesh_approximation_plugin.cpp [hunk 3]
```diff
@@ -508,19 +467,18 @@ void Polyhedron_demo_surface_mesh_approximation_plugin::on_buttonTeleport_clicke
...
```
Comment (maxGimeno) [inline Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/Surface_mesh_approximation_plugin.cpp]: And here.

File: Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/Surface_mesh_approximation_plugin.cpp [hunk 4]
```diff
@@ -562,19 +520,18 @@ void Polyhedron_demo_surface_mesh_approximation_plugin::on_buttonSplit_clicked()
...
```
Comment (maxGimeno) [inline Polyhedron/demo/Polyhedron/Plugins/Surface_mesh/Surface_mesh_approximation_plugin.cpp]: And here.

**Comment by maxGimeno** [comment]
I took the liberty of fixing the errors with the new master.

**Comment by maxGimeno** [comment]
Successfully tested in https://cgal.geometryfactory.com/CGAL/testsuite/results-4.14-Ic-116.shtml

**Pull request merged**